src/stdatamodels/asdf_in_fits.py

```diff
@@ -39,20 +40,37 @@ def open(filename_or_hdu, **kwargs):  # noqa: A001
         Filename of the FITS file or an open `astropy.io.fits.HDUList`
         containing the ASDF data. If a filename is provided it
         will be opened with :func:`astropy.io.fits.open`.
+    ignore_missing_extensions : bool, optional
+        If `True`, ignore missing extensions in the FITS file.
+        Defaults to `False`.
+    ignore_unrecognized_tag : bool, optional
+        If `True`, ignore unrecognized tags in the ASDF data.
+        Defaults to `False`.
     **kwargs
-        Additional keyword arguments to pass to :func:`asdf.open`
+        Additional keyword arguments to pass to asdf.open.
+        Usage of kwargs is deprecated and will be removed in a future version.
```

Comment:
tldr; looks good to me

When we remove this we will be dropping some functionality. For example passing a custom schema is currently supported via these kwargs only. That's the only possibly useful argument I see for `asdf.open` that's being handled by kwargs and I'm ok with dropping it (since this API is meant to be simple and to help folks migrating from `AsdfInFits`). If we hear anything about this `DeprecationWarning` we can always add `custom_schema` here. I'm in favor of not adding `custom_schema` here.

Reply:
Sounds good.  Yes this will remove some functionality, and I think it's therefore good that we are deprecating it.